résoud le bug de 404 sur le lien de unfollow (#1884)

artragis opened the pull request:
| Q | R |
| --- | --- |
| Correction de bugs ? | [oui] |
| Nouvelle Fonctionnalité ? | [non] |
| Tickets concernés | [#1884] |

Continué dans la #1901
# Pour la QA

Voici le scénario : 

lancez le site avec les fixtures des forums, des catégories et des sujets incluses.

Ensuite, connectez-vous avec n'importe quel utilisateur et allez suivre un sujet.

La résolution du bug, c'est à dire vérifier que lorsque l'on clique sur "ne plus suivre" dans la side bar, tout fonctionne,  est à vérifier depuis plusieurs pages : 
- l'index du forum (/forums)
- un sujet tiers (/forums/sujet/id/slug/)
- l'index d'un sous forum (/forums/slug_cat/slug_forum/)
- l'index d'une catégorie (/forums/slug_cat/)

Eskimon committed:
ZEP-03: Initialisation

Eskimon committed:
ZEP-03: Ajout documentation technico-fonctionnelle

Eskimon committed:
ZEP-03: Ajout des tests unitaires

Eskimon committed:
ZEP-03: Ajout des modifications proposées

Eskimon committed:
ZEP-03: Ajout de la pagination

Eskimon committed:
ZEP-03: Corrige un test et améliore une requete

Eskimon committed:
ZEP-03: Petite amélioration d'un texte

Eskimon committed:
ZEP-03: Plein de petites corrections

Eskimon committed:
ZEP-03: Ajout d'images fixtures et evo template

artragis committed:
typo dans le nom d'une constante

artragis committed:
trvail sur les factory

Eskimon committed:
Update doc, template, images fixtures

artragis committed:
On peut charger un media grâce aux fixtures

artragis committed:
la fixture media

artragis committed:
documentation

sandhose committed:
Update des dépendances NPM

artragis committed:
Le readme est à jour

artragis committed:
Factory pour les contenus à validation. Simplification des tests.

artragis committed:
typo dans la doc

Commit:
correction du design de la page d'aide

Commit:
passage sur une colonne

Eskimon committed:
Correction rebase + PEP

poulp committed:
ajout titre defaut aux nouveaux mp collaboration

Eskimon committed:
Petites améliorations de design

Eskimon committed:
Internationalisation

Conflicts:

	conf/locale/en/LC_MESSAGES/django.po

poulp committed:
ajout test mp GET title

poulp committed:
fix pep8

Eskimon committed:
Repare des tests

Eskimon committed:
Petite maj+typo dans la doc

Eskimon committed:
Maj de la doc de déploiement

Eskimon committed:
Correction de fautes

Eskimon committed:
Fixtures d'artragis et minifix front

Conflicts:

	conf/locale/en/LC_MESSAGES/django.po

Eskimon committed:
Gestion du cas ou media/ est absent

steenzout committed:
mp/views.py : nouvelle condition except pour le cas ou la conversation n'existe pas.

Eskimon committed:
Rebase + correction de render

artragis committed:
todo AJAX staff enelvé car non pertinant

artragis committed:
clarifie la pagination

artragis committed:
renomme la fonction get_users en get_linked_users qui est plus parlant.

artragis committed:
nettoie les todo déjà corrigé mais laissés en commentaire

Eskimon committed:
Corrections sur le template/front

Eskimon committed:
Internationalise et rend la demande d'aide plus clair

Conflicts:

	conf/locale/en/LC_MESSAGES/django.po

Cydonia7 committed:
Plus de duplicata dans la liste de l'autocomplétion

antoyo committed:
Fixe l’issue #1880 (mauvais lien dans MP).

artragis committed:
résoud le bug de 404 sur le lien de unfollow

artragis committed:
documentation du suivi des sujets sur le forum

gustavi commented on zds/forum/views.py:
```diff
@@ -382,9 +386,11 @@ def edit(request):
+        except ValueError:
             # problem in variable format
             raise Http404
+        except:
```
BEURK ! On peut me mettre une belle exception ici ? Pour moi **à na pas merger** tant qu'on a ça !

(Oui je sais, ça ne vient pas de toi mais autant faire des choses propres)

artragis replied:
Je vais vérifier mais à mon avis je n'en ai même pas besoin.

artragis committed:
définit l'exception pour raise

Situphen committed:
Les longues signatures ne cassent plus le design

pierre-24 commented on doc/sphinx/source/forum/forum.rst:
```diff
@@ -74,3 +74,28 @@ Il existe actuellement 3 filtres pour filtrer les sujets dans un forum :
+
+Nous permettons à nos membres de "suivre" un sujet directement sur le site. "Suivre un sujet", cela signifie que, lorsqu'un nouveau message est posté sur ce sujet, l'icône de notification nous en informe.
+
+Dès lors vous pourrez par un simple clic allé au dernier message non lu.
```
*aller :)

pierre-24 commented:
Rapport de QA (bordel, ça fait des millénaires) : 
- De l'index de forum : OK
- D'un sujet tiers : **pas OK**
- D'un sous-forum : **pas OK**
- D'une catégorie : OK

Donc résolu à moitié.

gustavi committed:
Fix #1891 : correction de balises « url » fausses

pierre-24 committed:
Correction du lien de citation sur un article

(remplacer `topic.pk` par `article.pk` dans `templates/article/reaction/new.html`)

Eskimon committed:
Ameliorations design par Situphen

artragis committed:
fix do not follow dans les listes de topics et les sujets

pierre-24 commented:
@artragis : tu viens de changer quelque chose ou c'est juste un rebase bizarre, là ?

artragis commented:
rebase foireux, je refais une branche et donc une PR à partir de dev. Mais le merge freeze risque de ralentir le tout.
